Monofoxe/Pipefoxe/SpriteGroup/SpriteGroupMathParser.cs

```diff
@@ -0,0 +1,228 @@
+	public class SpriteGroupMathParser
+	{
+
+		public SpriteGroupMathParser(RawSprite sprite)
+		{
+			_rawSprite = sprite; // Maybe this should not be here but hey... Fox are sneaky 
+		}
+
+		SpriteGroupMathTokenizer _tokenizer;
+		RawSprite _rawSprite;
+		bool _yAxis = false; // Yes, constant center made me do this, why, because I am awesome
+
+		public int Parse(string text, bool yAxis = false)
+		{
+			_tokenizer = new SpriteGroupMathTokenizer(text);
+			_yAxis = yAxis;
+			var expr = ParseAddSubstract();
+
+			if (_tokenizer.CurrentToken != Token.EOF)
+			{
+				throw new Exception("Unexpected characters at end of expression");
+			}
+			return expr.Eval();
+		}
+		/// <summary>
+		/// Handles Unary expresions, this let us negate expresions (even be redundant like 10 + +1)
+		/// It reuses the <seealso cref="Token.Sub"/> and <seealso cref="Token.Sum"/> to handle it
+		/// </summary>
+		/// <returns>The Unary expresion reduced</returns>
+		Node ParseUnary()
+		{
+			switch (_tokenizer.CurrentToken)
+			{
+				case Token.Sum:
+					_tokenizer.NextToken();
+					return ParseUnary();
+				case Token.Sub:
+					_tokenizer.NextToken();
+
+					Node rhs = ParseUnary(); // recursive methods are fun, this let us be as redundant as we want
+
+					return new UnaryOperationNode(rhs, a => -a);
+			}
+			return ParseLeaf();
+		}
+		/// <summary>
+		/// Handles sums and substraction, it follows PEMDAS by asking if their left hand side and right hand side are multiplication
+		/// </summary>
+		/// <returns></returns>
+		Node ParseAddSubstract()
+		{
+			Node lhs = ParseMultiplyDivide();
+
+			while (true)
+			{
+				Func<int, int, int> op = null;
+				switch (_tokenizer.CurrentToken)
+				{
+					case Token.Sum:
+						op = (a, b) => a + b;
+						break;
+					case Token.Sub:
+						op = (a, b) => a - b;
+						break;
+				}
+				if (op == null)
+				{
+					return lhs;
+				}
+				_tokenizer.NextToken();
+
+				Node rhs = ParseMultiplyDivide();
+				lhs = new BinaryOperationNode(lhs, rhs, op);
+			}
+
+		}
+		/// <summary>
+		/// Resolves Multiplication and division in PEMDAS order, let <see cref="ParseUnary"/> resolve negatives and such expresions
+		/// </summary>
+		/// <returns></returns>
+		Node ParseMultiplyDivide()
+		{
+			Node lhs = ParseUnary();
+
+			while (true)
+			{
+				Func<int, int, int> op = null;
+				switch (_tokenizer.CurrentToken)
+				{
+					case Token.Mul:
+						op = (a, b) => a * b;
+						break;
+					case Token.Div:
+						op = (a, b) => a / b;
+						break;
+				}
+				if (op == null)
+				{
+					return lhs;
+				}
+				_tokenizer.NextToken();
+
+				Node rhs = ParseUnary();
+				lhs = new BinaryOperationNode(lhs, rhs, op);
+			}
+		}
+		/// <summary>
+		/// This recognize our terminal elements, like number and constants, also reduce parenthesis expresions
+		/// </summary>
+		/// <returns>Terminal element</returns>
+		Node ParseLeaf()
+		{
+			switch (_tokenizer.CurrentToken)
+			{
+				case Token.Number:
```

Comment:
This is barely readable. Add braces to cases, or better yet - get rid of the switch.

Reply:
Decided first form, not a fan of 
```csharp
if(_tokenizer.CurrentToken == Token.Number) 
{
	//...
}
```
with a lot of conditions